ui/components/app/signature-request/signature-request.component.js

```diff
@@ -61,9 +72,56 @@ export default class SignatureRequest extends PureComponent {
     )}`;
   }
 
+  getNetworkName() {
+    const { provider } = this.props;
+    const providerName = provider.type;
+    const { t } = this.context;
+
+    switch (providerName) {
+      case NETWORK_TYPES.MAINNET:
+        return t('mainnet');
+      case NETWORK_TYPES.GOERLI:
+        return t('goerli');
+      case NETWORK_TYPES.SEPOLIA:
+        return t('sepolia');
+      case NETWORK_TYPES.LOCALHOST:
+        return t('localhost');
+      default:
+        return provider.nickname || t('unknownNetwork');
+    }
+  }
+
+  renderHeader = () => {
+    const { conversionRate, nativeCurrency } = this.props;
+    const {
+      fromAccount: { address, balance, name },
+    } = this.props;
+
+    const currentNetwork = this.getNetworkName();
+
+    const balanceInBaseAsset = conversionUtil(balance, {
+      fromNumericBase: 'hex',
+      toNumericBase: 'dec',
+      fromDenomination: 'WEI',
+      numberOfDecimals: 6,
+      conversionRate,
+    });
+
+    return (
+      <div className="request-signature__account">
+        <NetworkAccountBalanceHeader
+          networkName={currentNetwork}
+          accountName={name}
+          accountBalance={balanceInBaseAsset}
+          tokenName={nativeCurrency}
+          accountAddress={address}
+        />
+      </div>
+    );
+  };
```

Comment:
We should avoid extending the pattern of having functions/methods that return portions of the react tree, please have this rendered inline with the other jsx

Reply:
done